proud-puffins/djangoProject/earlydating/views.py

```diff
@@ -0,0 +1,204 @@
+            return redirect('earlydating-login')
+    context = {'form': form}
+    return render(request, 'dating/register.html', context)
+
+
+def logoutUser(request):
+    logout(request)
+    return redirect('earlydating-login')
+
+
+def about(request):
+    return render(request, 'dating/about.html')
+
+
+@login_required(login_url='earlydating-login')
+@allowed_users(allowed_roles=['profile'])
+def editprofile(request):
+    if request.method == 'POST':
+        u_form = UserUpdateForm(request.POST, instance=request.user)
+        p_form = ProfileUpdateForm(request.POST, request.FILES, instance=request.user.profile)
+
+        if u_form.is_valid() and p_form.is_valid():
+            u_form.save()
+            p_form.save()
+            messages.success(request, 'Your profile has been updated.')
+            return redirect('../YourProfile')
+
+    else:
+        u_form = UserUpdateForm(instance=request.user)
+        p_form = ProfileUpdateForm(instance=request.user.profile)
+
+    context = {
+        'u_form': u_form,
+        'p_form': p_form
+    }
+
+    return render(request, 'dating/edit_profile.html', context)
+
+
+@login_required(login_url='earlydating-login')
+@allowed_users(allowed_roles=['profile'])
+def DateMatcher(request):
+    logged_user = request.user
+    if request.method == 'POST':
+        print(request.POST)
+        if like_pk := request.POST.get('Like'):
+            voted = User.objects.get(pk=like_pk)
+            vote, _ = UserVote.objects.get_or_create(user=voted, voter=logged_user)
+            vote.vote = True
+        elif like_pk := request.POST.get('Unlike'):
+            voted = User.objects.get(pk=like_pk)
+            vote, _ = UserVote.objects.get_or_create(user=voted, voter=logged_user)
+            vote.vote = False
+        vote.save()
+        return redirect('earlydating-profile', pk=like_pk)
+    elif request.method == 'GET':
+        user = get_unvoted(logged_user)[0]
+        return redirect('earlydating-profile', pk=user.pk)
+
+
+def getfilters(user):
+    u_sex = user.profile.sex
+    gender = ['Male', 'Female']
+    try:
+        if user.profile.preference == 'straight':
+            Q1 = Q(profile__sex=gender[int(not gender.index(u_sex))])
+            Q2 = Q(profile__preference__in=['straight', 'bisexual'])
+            return Q1 & Q2
+
+        elif user.profile.preference == 'gay':
+            Q1 = Q(profile__sex=u_sex)
+            Q2 = Q(profile__preference__in=['gay', 'bisexual'])
+            return Q1 & Q2
+
+        elif user.profile.preference == 'bisexual':
+            Q1 = Q(profile__sex=u_sex)
+            Q2 = Q(profile__preference='gay')
+            Q3 = Q(profile__sex=gender[int(not gender.index(u_sex))])
+            Q4 = Q(profile__preference='straight')
+            Q5 = Q(profile__preference='bisexual')
+            return (Q1 & Q2) | (Q3 & Q4) | Q5
+
+    except Exception as e:
+        print(e)
+        return None
```

Comment:
Typically you will want to have as few lines as possible in a try block, because you want to be aware of which lines are likely to cause *expected* exceptions and handle only the ones you expect. In circumstances where you're sure you need to handle a broad number of exceptions that could be raised across several lines, you will still at least want to specify which exceptions you expect, and ensure that the error message is logged rather than simply printed. The logging module in the standard library provides that functionality.

Even though I haven't used Django personally, I feel that I can intuitively understand what `getfilters` is for just based on how it's written. I'm a bit confused by `int(not gender.index(u_sex))`, however. I would avoid writing code that relies on the fact that `True` and `False` are `1` and `0` unless that's imposed by another library because it's less communicative.